scripts/generate-icons.js

```diff
@@ -0,0 +1,124 @@
+    .join('');
+}
+
+/**
+ * Generate source for a Preact icon component from its SVG source.
+ *
+ * @param {string} name - Name of the Preact component
+ * @param {string} src - SVG source
+ * @param {string} inputFileName - Name of the file that `src` comes from.
+ *   This is used in documentation only.
+ */
+function generateIcon(name, src, inputFileName) {
+  const optimized = optimize(src, {
+    plugins: [
+      {
+        name: 'preset-default',
+        params: {
+          overrides: {
+            // View boxes are needed for some icons to be able to render
+            // them smaller than their width and height values
+            removeViewBox: false,
+          },
+        },
+      },
+    ],
+  });
+
+  // Take the body of the SVG and just treat it as JSX. This works because
+  // SVG is XML (so all tags are closed) and SVG attribute names don't need
+  // to be changed for Preact JSX.
+  //
+  // A caveat with this approach is that all attribute values will be strings.
+  // We assume the performance cost of this is very minor.
+  const content = optimized.data.replace(
+    /<svg ([^>]+)>/,
+    '<svg $1 {...props}>'
+  );
+
+  const jsx = `
+${AUTO_GENERATE_COMMENT}
+
+/**
+ * Icon generated from ${inputFileName}
+ *
+ * @param {import('preact').JSX.SVGAttributes<SVGSVGElement>} props
+ */
+export default function ${name}(props) {
+  return ${content};
+}
+`;
+  return format(jsx, { parser: 'babel' });
+}
+
+/**
+ * Generate an icon component .js file from an SVG input file.
+ *
+ * The component name will be generated by pascal-casing the input file name
+ * and appending "Icon". eg. "copy-file.svg" => `CopyFileIcon`.
+ *
+ * @param {string} inputFile - Input SVG file path
+ * @param {string} outputDir - Output directory
+ */
+function generateIconFromFile(inputFile, outputDir) {
+  const src = readFileSync(inputFile);
+  const basename = path.basename(inputFile).replace(/\.svg/, '');
+
+  const iconName = kebabCaseToPascalCase(basename);
+  const componentName = iconName + 'Icon';
+
+  const outputFile = `${outputDir}/${iconName}.js`;
+  const outputSrc = generateIcon(componentName, src, basename + '.svg');
+
+  writeFileSync(outputFile, outputSrc);
+}
+
+/**
+ * Generate an index module for a set of (icon) component modules.
+ * This will create an export for every file ending in `.js` in `componentDir`.
+ * This assumes that every module has a single default export.
+ *
+ * @param {string} componentDir - Directory containing components
+ */
+function generateIconIndex(componentDir) {
+  const iconComponents = readdirSync(componentDir).filter(
+    file => file.endsWith('.js') && !file.includes('index')
+  );
+  let outputSrc = `${AUTO_GENERATE_COMMENT}\n`;
+  for (let componentFile of iconComponents) {
+    const componentModule = path.basename(componentFile, '.js');
+    const componentName = `${componentModule}Icon`;
+    outputSrc += `export { default as ${componentName} } from './${componentModule}';\n`;
+  }
```

Comment:
This was a game of "this is way to freaking hard" that I walked away from as running this content through `prettier.format` was throwing up resistance in annoying ways. That is, I did set out to run this through prettier, but ran into obstacles:

* Running this content through `prettier.format` using the same mechanism as for the individual icon components — using `parser: 'babel'` but otherwise using defaults — resulted in double quotes in the output, which breaks linting.
* Prettier has a [mechanism for resolving config](https://prettier.io/docs/en/api.html#prettierresolveconfigfilepath--options) in their API that can then be used as `options` for `format` — with the hope that passing that would get us some single quotes — but I can't win the importing game after a few attempts (module mismatch):

```
import { format, resolveConfig } from 'prettier';
                 ^^^^^^^^^^^^^
SyntaxError: Named export 'resolveConfig' not found. The requested module 'prettier' is a CommonJS module, which may not support all module.exports as named exports.
CommonJS modules can always be imported via the default export, for example using:
```

I tried a couple of different ways to import `resolveConfig` to no avail and gave up because this _really doesn't seem worth it_ as the output already passes linting without any prettify-ing. Happy to re-assess, but sheesh. :)